#### Subscribers ########
subscribers:
  # debug_rgb:
  #   fusion: ['image_color']
  #   topic_name_camera: /zed2i/zed_node/left/image_rect_color
  #   topic_name_camera_info: /zed2i/zed_node/right/camera_info
  #   channels: ["rgb_image"]    
  #   data_type: image
  #   fusion_types:
  #     array_fusion: []
  #     channels: []

  # debug_rgb_3_seperate:
  #   fusion: ['image_exponential','image_exponential','image_exponential']
  #   topic_name_camera: /zed2i/zed_node/right/image
  #   topic_name_camera_info: /zed2i/zed_node/right/camera_info
  #   channels: ["r_channel", "g_channel", "b_channel"]    
  #   data_type: image
  
  # debug_rgb_2_seperate:
  #   fusion: ['image_exponential', 'image_exponential']
  #   topic_name_camera: /zed2i/zed_node/left/image_rect_color
  #   topic_name_camera_info: /zed2i/zed_node/right/camera_info
  #   channels: ["r_channel", "g_channel"]    
  #   data_type: image

  # debug_mono_exponential:
  #   fusion: ['image_exponential']
  #   topic_name_camera: /debug_image
  #   topic_name_camera_info: /zed2i/zed_node/right/camera_info
  #   channels: ["single_channel_semantic"]    
  #   data_type: image
    
  # front_cam:
  #   channels: ['rgb']
  #   fusion: ['color']
  #   topic_name: '/elevation_mapping/pointcloud_semantic'
  #   data_type: pointcloud
  
  alphasense_front_rgb:
    fusion: ['image_color']
    topic_name_camera: /alphasense_driver_ros/cam4/debayered
    topic_name_camera_info: /alphasense_driver_ros/cam4/camera_info
    channels: ["rgb_image"]
    data_type: image

  
  alphasense_left_rgb:
    fusion: ['image_color']
    topic_name_camera: /alphasense_driver_ros/cam3/debayered
    topic_name_camera_info: /alphasense_driver_ros/cam3/camera_info
    channels: ["rgb_image"]
    data_type: image


  alphasense_right_rgb:
    fusion: ['image_color']
    topic_name_camera: /alphasense_driver_ros/cam5/debayered
    topic_name_camera_info: /alphasense_driver_ros/cam5/camera_info
    channels: ["rgb_image"]
    data_type: image


  wvn_prediction:
    fusion: ['image_exponential']
    topic_name_camera: '/wild_visual_navigation_node/traversability_raw'
    topic_name_camera_info: '/wild_visual_navigation_node/camera_info'
    channels: ["visual_traversability"]
    data_type: image
    
  # front_cam:
  #   channels: ['rgb','feat_0','feat_1','grass','tree','fence','person']
  #   fusion: ['color','average','average','class_average','class_average','class_average','class_average']
  #   topic_name: '/elvation_mapping/pointcloud_semantic'
  #   semantic_segmentation: False
  #   segmentation_model: 'detectron_coco_panoptic_fpn_R_101_3x'
  #   feature_config:
  #     name: 'DINO'
  #     interpolation: 'bilinear'
  #     model: "vit_small"
  #     patch_size:  16
  #     dim: 5
  #     dropout: False
  #     dino_feat_type: "feat"
  #     input_size: [80, 160]
  #     projection_type: "nonlinear"

  #   cam_info_topic: "/camera/depth/camera_info"
  #   image_topic: "/camera/rgb/image_raw"
  #   depth_topic: "/camera/depth/image_raw"
  #   cam_frame: camera_rgb_optical_frame
  #   confidence: False
  #   confidence_topic: "/camera/depth/image_raw"
  #   confidence_threshold: 10
  #   feature_extractor: False
    
  # front_bpearl:
  #   channels: []
  #   fusion: []
  #   topic_name: /point_cloud_filter_rsl/filter_and_merger_rsl
  #   data_type: pointcloud
  
  # rear_bpearl:
  #   channels: []
  #   fusion: []
  #   topic_name: /robot_self_filter/bpearl_rear/point_cloud
  #   data_type: pointcloud
  
  front_depth:
    channels: []
    fusion: []
    topic_name: /depth_camera_front/point_cloud_self_filtered
    data_type: pointcloud

  rear_depth:
    channels: []
    fusion: []
    topic_name: /depth_camera_rear/point_cloud_self_filtered
    data_type: pointcloud

  left_depth:
    channels: []
    fusion: []
    topic_name: /depth_camera_left/point_cloud_self_filtered
    data_type: pointcloud

  right_depth:
    channels: []
    fusion: []
    topic_name: /depth_camera_right/point_cloud_self_filtered
    data_type: pointcloud
  
  velodyne:
    channels: []
    fusion: []
    topic_name: /point_cloud_filter/lidar/point_cloud_filtered
    data_type: pointcloud
